### ISSUE TITLE
Add actor to modifyTokenAttribute hook

### ISSUE THREAD

**Issue by KSops** [foundryvtt/foundryvtt#11537]
### User Experience

Request: Add the actor context to the set of parameters passed to the modifyTokenAttribute hook

Experience:
Currently the `modifyTokenAttribute` hook callback only allows for accessing original method param data but without the actor context. Without knowing attributes on the actor, this severely limits the ability to override logic with this hook.

e.g. The target actor has HP 100/50 and takes 1 HP damage; the result will be 50 HP due to the clamping logic instead of 99 HP.

I'm not proposing to change this logic, since there's many cases where this *is* desirable. But I am proposing to allow for hooks to have access to sufficient data to alter such behaviours. Without the actor context I don't believe this is achievable.

**Comment by aaclayton** [comment]
I agree that the current design is not great. We should pass the Actor in also.
